controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -259,6 +267,99 @@ func (r *AKODeploymentConfigReconciler) reconcileCloudUsableNetwork(
 	return res, nil
 }
 
+func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
+	ctx context.Context,
+	log logr.Logger,
+	adc *akoov1alpha1.AKODeploymentConfig,
+) (ctrl.Result, error) {
+	res := ctrl.Result{}
+
+	log.Info("Start reconciling AVIInfraSetting")
+
+	if adc.Spec.ControlPlaneNetwork.Name == "" {
+		log.Info("ControlPlaneNetwork is empty in akoDeploymentConfig, skip creating AVIInfraSetting")
+		return res, nil
+	}
+
+	newAviInfraSetting := r.createAviInfraSetting(adc)
+
+	if err := r.Get(ctx, client.ObjectKey{
+		Name: haprovider.GetAviInfraSettingName(adc),
+	}, newAviInfraSetting); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("AVIInfraSetting doesn't exist, start creating it")
+			return res, r.Create(ctx, newAviInfraSetting)
+		}
+		log.Error(err, "Failed to get AVIInfraSetting, requeue")
+		return res, err
+	}
+	aviInfraSetting := newAviInfraSetting.DeepCopy()
+	return res, r.Update(ctx, aviInfraSetting)
+}
+
+func (r *AKODeploymentConfigReconciler) createAviInfraSetting(adc *akoov1alpha1.AKODeploymentConfig) *akov1alpha1.AviInfraSetting {
+
+	return &akov1alpha1.AviInfraSetting{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AviInfraSetting",
+			APIVersion: "ako.vmware.com/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: haprovider.GetAviInfraSettingName(adc),
+		},
+		Spec: akov1alpha1.AviInfraSettingSpec{
+			SeGroup: akov1alpha1.AviInfraSettingSeGroup{
+				Name: adc.Spec.ServiceEngineGroup,
+			},
+			Network: akov1alpha1.AviInfraSettingNetwork{
+				VipNetworks: []akov1alpha1.AviInfraSettingVipNetwork{{
+					NetworkName: adc.Spec.ControlPlaneNetwork.Name,
+					Cidr:        adc.Spec.ControlPlaneNetwork.CIDR,
+				}},
+			},
+			// TODO: expose ShardSize in akoDeploymentConfig
+			L7Settings: akov1alpha1.AviInfraL7Settings{
+				ShardSize: "MEDIUM",
```

Comment:
Why hardcode this one?
This is already added in `adc.spec.AKOIngressConfig.ShardVSSize`

Reply:
Thanks! Didn't notice it before.